projects/plugins/wpcomsh/feature-plugins/blaze.php

```diff
@@ -61,5 +63,51 @@ function wpcomsh_activate_blaze_module_on_launching( $old_value, $new_value ) {
 	if ( $blog_public === 1 ) {
 		wpcomsh_activate_blaze_module();
 	}
+
+	return $new_value;
 }
 add_filter( 'update_option_blog_public', 'wpcomsh_activate_blaze_module_on_launching', 10, 2 );
+
+/**
+ * Delete the transient for the given site id.
+ *
+ * @return void
+ */
+function wpcomsh_blaze_purge_transient_cache() {
+	$transient = 'jetpack_blaze_site_supports_blaze_' . Jetpack_Connection::get_site_id();
+	delete_transient( $transient );
+}
+
+/**
+ * Delete the caching transient when coming soon is changed.
+ */
+add_action(
+	'pre_update_option_wpcom_public_coming_soon',
+	function ( $option ) {
+		wpcomsh_blaze_purge_transient_cache();
+		return $option;
+	}
+);
+
+/**
+ * Delete the caching transient when the blog visibility option changes.
+ */
+add_action(
+	'pre_update_option_blog_public',
+	function ( $option ) {
+		wpcomsh_blaze_purge_transient_cache();
+		return $option;
+	}
+);
+
+/**
+ * On Atomic sites the Promote with Blaze option is enabled.
+ *
+ * @phan-suppress PhanUndeclaredFunctionInCallable -- jetpack_blaze_post_row_actions_disable is part of jetpack.
```

Comment:
I think the recommended way to solve is by explicitly declaring the dependencies. See https://github.com/Automattic/jetpack/blob/f52f223ae2f9115e4dbbfb58b6b513da83953c75/projects/plugins/wpcomsh/.phan/config.php#L24-L26

> DO NOT add references to files in other packages like this! Generally packages should be listed in composer.json 'require'. If there are truly optional dependencies or circular dependencies that can't be cleaned up, one package may list the other in 'require-dev' and `extra.dependencies.test-only' instead. See packages/config for an example.


I recently had to do something similar here: https://github.com/Automattic/jetpack/pull/40479/files/e4cb7c7fb0e8c3d88cbc2e0d65fb2d42b24c013c..e0bc71610c0842375e32f5887d1dfafd3655e183